File: be/src/main/java/sg/flow/services/BankQueryServices/IBankQueryService.java
package sg.flow.services.BankQueryServices;

import java.time.LocalDate;

import sg.flow.entities.Bank;
import sg.flow.models.transaction.TransactionHistoryList;

public interface IBankQueryService {
    public TransactionHistoryList getTransactionHistoryBetween(int userId, String accountNumber, LocalDate startDate, LocalDate endDate);

    public boolean hasAccountNumber(String accountNumber);

    public Bank getBank();
}

File: be/src/main/kotlin/sg/flow/services/BankQueryServices/IBankQueryService.kt
package sg.flow.services.BankQueryServices

import sg.flow.entities.Bank
import sg.flow.models.transaction.TransactionHistoryList
import java.time.LocalDate

interface IBankQueryService {
    suspend fun getTransactionHistoryBetween(
        userId: Int, 
        accountNumber: String, 
        startDate: LocalDate, 
        endDate: LocalDate
    ): TransactionHistoryList
    
    suspend fun hasAccountNumber(accountNumber: String): Boolean
    suspend fun getBank(): Bank
} 